contracts/RootChain.sol

```diff
@@ -356,7 +356,8 @@ contract RootChain {
         require(exits[exitId].amount == 0);
 
         InFlightExit storage inFlightExit = _getInFlightExit(_outputTx);
-        if (!flagged(inFlightExit.exitMap)) {
+        // Check whether IFE is either ongoing or finished
+        if (inFlightExit.exitStartTimestamp != 0 || flagged(inFlightExit.exitMap)) {
             // Check if this output was piggybacked or exited in an in-flight exit
             require(!inFlightExit.exitMap.bitSet(oindex + MAX_INPUTS) && !inFlightExit.exitMap.bitSet(oindex + MAX_INPUTS + MAX_INPUTS*2));
             // Prevent future piggybacks on this output
```

Comment:
@pik694 actually my previous comment was more about this line. The relationship of the action here and the condition seems unclear to me at first glance. Now I guess we just flag the output as "exited" anyway as it won't hurt even for a "finalized" exit it would never gonna use this change, right?

Reply:
Flagging an output as exited ensures that none can piggyback it in the future.

Reply:
yes, but if I understand correct, once it is `flagged(inFlightExit.exitMap)` then the whole IFE is finalized so nothing can further piggyback on isn't it?

Reply:
In this case, that's right, but keep in mind that the flow is a little more complicated here.